R/zzz.R

```diff
@@ -1,5 +1,27 @@
 .lockbox_env <- new.env()
 
+.get_safely_from_envir <- function(target, default = NULL) {
+  res <- tryCatch(
+    get(target, envir = .lockbox_env)
+  , error = function(e) {default})
+}
+
+set_session_id <- function() {
+  if (is.null(.get_safely_from_envir("session_id"))) {
+    if (requireNamespace("uuid", quietly = TRUE)) {
+      assign("session_id", uuid::UUIDgenerate(), envir = .lockbox_env)
+    } else {
+      assign("session_id", digest::digest(as.integer(Sys.time())), envir = .lockbox_env)
+    }
+  }
+  
+  # We don't want to clutter up ~/.R/ with tons of symlinked dirs, lets try to do some housekeeping
+  reg.finalizer(.lockbox_env, function(env) {
```

Comment:
Yeah this is a good question. I just tested [unload hooks](https://stat.ethz.ch/R-manual/R-devel/library/base/html/ns-hooks.html) and it doesn't look like they run if you `q()` directly. I'll do some testing to ensure this covers most sane cases of the R interpreter exiting.